File: app/src/main/java/mohit/codex_iter/www/awol/application/CleanCacheApplication.java
package mohit.codex_iter.www.awol.application;
import android.app.Application;

import java.io.File;

public class CleanCacheApplication extends Application {

    private static CleanCacheApplication instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }

    public static CleanCacheApplication getInstance() {
        return instance;
    }

    public void clearApplicationData() {
        File cacheDirectory = getCacheDir();
        File applicationDirectory = new File(cacheDirectory.getParent());
        if (applicationDirectory.exists()) {
            String[] fileNames = applicationDirectory.list();
            for (String fileName : fileNames) {
                if (!fileName.equals("lib")) {
                    deleteFile(new File(applicationDirectory, fileName));
                }
            }
        }
    }

    public static boolean deleteFile(File file) {
        boolean deletedAll = true;
        if (file != null) {
            if (file.isDirectory()) {
                String[] children = file.list();
                for (int i = 0; i < children.length; i++) {
                    deletedAll = deleteFile(new File(file, children[i])) && deletedAll;
                }
            } else {
                deletedAll = file.delete();
            }
        }

        return deletedAll;
    }

}

File: app/src/main/java/codex/codex_iter/www/awol/application/CleanCacheApplication.kt
package codex.codex_iter.www.awol.application

import android.app.Application
import java.io.File
import java.util.*

class CleanCacheApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    fun clearApplicationData() {
        val cacheDirectory = cacheDir
        val applicationDirectory = File(Objects.requireNonNull(cacheDirectory.parent))
        if (applicationDirectory.exists()) {
            val fileNames = applicationDirectory.list()
            if (fileNames != null) {
                for (fileName in fileNames) {
                    if (fileName != "lib") {
                        deleteFile(File(applicationDirectory, fileName))
                    }
                }
            }
        }
    }

    companion object {
        var instance: CleanCacheApplication? = null
            private set

        fun deleteFile(file: File?): Boolean {
            var deletedAll = true
            if (file != null) {
                if (file.isDirectory) {
                    val children = file.list()
                    if (children != null) {
                        for (child in children) {
                            deletedAll = deleteFile(File(file, child)) && deletedAll
                        }
                    }
                } else {
                    deletedAll = file.delete()
                }
            }
            return deletedAll
        }
    }
}